### PR TITLE
run rustfmt on compiletest folder in src/tools/ folder

### PR DESCRIPTION
**Pull request by srinivasreddy**

### CODE REVIEW

**Comment by srinivasreddy** [comment]
cc @nrc

File: src/tools/compiletest/src/json.rs [hunk 1]
```diff
@@ -87,16 +89,15 @@ fn push_expected_errors(expected_errors: &mut Vec<Error>,
...
-                          .filter(|span| span.is_primary)
-                          .collect();
+    let spans_in_this_file: Vec<_> = diagnostic.spans
+        .iter()
+        .filter(|span| Path::new(&span.file_name) == Path::new(&file_name))
+        .collect();
+
+    let primary_spans: Vec<_> = spans_in_this_file.iter()
+        .cloned()
+        .filter(|span| span.is_primary)
```
Comment (nikomatsakis) [inline src/tools/compiletest/src/json.rs]: @nrc how does rustfmt decide when to align to `.` in a method-call chain? Or does it just never do it?

Comment (nrc) [inline src/tools/compiletest/src/json.rs]: It uses block indent atm. This has been something of a tricky point - we used to use visual indent which looked pretty but caused pretty bad rightward drift and caused problems with closures, so now we always use block indent which doesn't look as good, but does work well with others. Hopefully we can improve it further in the future

**Pull request merged**